src/socket.rs

```diff
@@ -1017,6 +1119,28 @@ impl Drop for Socket {
     }
 }
 
+/// Fix-up a created socket by setting the proper portability flags.
+#[allow(unused)]
+fn fixup_socket(mut sock: Socket) -> io::Result<Socket> {
+    #[cfg(windows)]
+    {
+        // Even though new sockets will have the no-inherit flag set,
+        // sockets obtained through [`Socket::accept_raw()`] are
+        // still inheritable on Windows.
+        sock._set_no_inherit(true)?;
+    }
+    #[cfg(unix)]
```

Comment:
Too broad, needs to be the inverse of `Type::set_closexec`. Don't want to make unneeded system calls.

Reply:
Which syscall here is unneeded?

Reply:
Ah, I see. This was never called on platforms that don't need to set cloexec separately. But sure, we can always call it and move the check in here.